Makefile

```diff
@@ -139,13 +142,61 @@ help-install:
 #
 ###############################################################
 
+### Deployment ###############################################
+#
+release-check:
+	@VERSION=$(shell sh -c "git describe --exact") && \
+	 if test -z "$$VERSION"; then                     \
+		echo VERSION is not set.;                 \
+		echo Please create a git tag;             \
+		exit 100;                                 \
+	else                                              \
+		echo "Releasing version: $$VERSION";      \
+	fi;
+
+release-prep: release-check
+	@VERSION=$(shell sh -c "git describe --exact") && \
+	python setup.py clean;                            \
+	python setup.py sdist;                            \
+	pip install dist/starfish-$$VERSION.tar.gz
+
+release-verify: release-check slow
+	@VERSION=$(shell sh -c "git describe --exact") && \
+	docker tag $(DOCKER_IMAGE) $(DOCKER_IMAGE):$$VERSION && \
+	docker tag $(DOCKER_IMAGE) $(DOCKER_IMAGE):$$VERSION-$(DOCKER_BUILD)
+
+release-upload: release-check
+	@VERSION=$(shell sh -c "git describe --exact") && \
+	printf '\n# Please execute the following steps\n';\
+	echo git push origin $$VERSION;                   \
+	echo docker push $(DOCKER_IMAGE);                 \
+	echo docker push $(DOCKER_IMAGE):$$VERSION;       \
+	echo docker push $(DOCKER_IMAGE):$$VERSION-$(DOCKER_BUILD);\
+	echo twine upload dist/starfish-$$VERSION.tar.gz
+
+clean:
+	rm -rf starfish.egg-info
+	rm -rf dist
+	rm -rf build
+	rm -rf .eggs
+	rm -f .cover*
+
+help-deployment:
+	$(call print_help, release-prep, Builds and installs the current tagged version)
+	$(call print_help, release-verify, Runs tests on the tagged version)
```

Comment:
Does `release-check` need to be documented?

Reply:
I intended it as more an internal target. A function call like `assert_tag` wouldn't suffice since it would run after other checks, e.g.:

```
release-verify: slow
    $(call assert_tag)
```